rotary/rules/rotary.smk

```diff
@@ -501,40 +524,38 @@ rule polish_polypolish:
 # TODO - the relative path workarounds in the shell here are a bit odd because polca outputs files in the present working directory
 rule polish_polca:
     input:
-        "polish/polypolish/polypolish.fasta"
+        qc_short_r1 = "{sample}/{sample}_R1.fastq.gz",
+        qc_short_r2 = "{sample}/{sample}_R2.fastq.gz",
+        polished = "{sample}/polish/polypolish/polypolish.fasta"
     output:
-        "polish/polca/polca.fasta",
-        temp("polish/polca/polypolish.fasta.unSorted.sam"),
-        temp("polish/polca/polypolish.fasta.alignSorted.bam")
+        polca_output = "{sample}/polish/polca/polca.fasta",
+        polypolish_sam = temp("{sample}/polish/polca/polypolish.fasta.unSorted.sam"),
+        polypolish_bam = temp("{sample}/polish/polca/polypolish.fasta.alignSorted.bam"),
     conda:
         "../envs/masurca.yaml"
     log:
-        "logs/polish/polca.log"
+        "{sample}/logs/polish/polca.log"
     benchmark:
-        "benchmarks/polish/polca.txt"
-    params:
-        qc_short_r1=SHORT_R1,
-        qc_short_r2=SHORT_R2,
-        outdir="polish/polca"
+        "{sample}/benchmarks/polish/polca.txt"
     threads:
         config.get("threads",1)
     resources:
         mem=config.get("memory")
     shell:
         """
-        cd {params.outdir}
-        polca.sh -a ../../{input} -r "{params.qc_short_r1} {params.qc_short_r2}" -t {threads} -m {resources.mem}G > ../../{log} 2>&1
-        ln -s "polypolish.fasta.PolcaCorrected.fa" "polca.fasta"
-        cd ../..
+        polca.sh -a {input.polished} -r "{input.qc_short_r1} {input.qc_short_r2}" -t {threads} -m {resources.mem}G > {log} 2>&1
```

Comment:
Sadly, POLCA dumps its output files in the directory where it is run. As of the last version I checked, there is no way to specify the output directory. This is why my old rule had some odd code in it to `cd` to the output directory before running the command. Unless the POLCA UI has been updated, we might have to keep that odd logic in the code

Reply:
@jmtsuji In the shell code below, I take a different approach. I run the POLCA, and it dumps the output to the CWD, as you mentioned. I then move (`mv`) these output files to the corrected directory afterwards. This avoids having to change the directory. It also allows the pathing to be easier to understand.

Reply:
@jmtsuji Does Polypolish make more output files than the ones listed?

Reply:
@jmtsuji I reverted back to your way of doing things. But I had to add `../` to all the paths to compensate for the additional `{sample}/` in the output directory move.

Reply:
@LeeBergstrand OK, sounds good!